pkg/catalogsourceconfig/registry.go

```diff
@@ -104,15 +106,51 @@ func (r *registry) ensureDeployment(operatorSources string) error {
 			return err
 		}
 		r.log.Infof("Created Deployment %s with registry command: %s", deployment.GetName(), registryCommand)
-	} else {
-		// Update the pod specification
+		return nil
+	}
+
+	// If the command has changed, we update the pod specification. If not, we kill the pod to
+	// force a re-sync with Quay. This is required so that we get updates from Quay when
+	// packages have not been added or removed from the spec.
+	if hasCommandChanged(deployment.Spec.Template.Spec.Containers[0].Command, registryCommand) {
 		deployment.Spec.Template = r.newPodTemplateSpec(registryCommand)
-		err = r.client.Update(context.TODO(), deployment)
+		err := r.client.Update(context.TODO(), deployment)
 		if err != nil {
 			r.log.Errorf("Failed to update Deployment %s : %v", deployment.GetName(), err)
 			return err
 		}
 		r.log.Infof("Updated Deployment %s with registry command: %s", deployment.GetName(), registryCommand)
+	} else {
+		options := &client.ListOptions{}
+		options.SetLabelSelector(r.getRawLabel())
+		pods := &core.PodList{}
+		err := r.client.List(context.TODO(), options, pods)
+		if err != nil {
+			r.log.Errorf("Failed to list pods in the Deployment %s: %v", deployment.GetName(), err)
+			return err
+		}
+
+		// There should only be one pod in the Deployment. If we find more
+		// than one we will delete the pod whose name starts with the
+		// CatalogSourceConfig's name
+		if len(pods.Items) > 1 {
+			r.log.Warnf("Found %d in Deployment %s. Will delete most likely match!",
+				len(pods.Items), deployment.GetName())
+		}
+
+		for _, pod := range pods.Items {
+			if !strings.HasPrefix(pod.GetName(), r.csc.GetName()) {
```

Comment:
Is this convention guaranteed when creating pods? Isn't it possible that the naming could change? Maybe it would be safer to use some unique identifier, like a label.

Reply:
Yeah there is no guarantee. I am using a label selector otherwise deployments won't work. I am just being paranoid that there could be another pod with the same label not created by us. Maybe I am being paranoid. Let me look into a better approach. Maybe scale down and scale up the deployment. It is just a pain to do that given we are not using `client-go`.